pkg/controller/capacity/capacity_controller.go

```diff
@@ -340,21 +355,23 @@ func (c *Controller) getSadPods(targetDeployment *appsv1.Deployment, clusterStat
 			fmt.Sprintf("there are %d sad pods", sadPodsCount))
 	}
 
-	return sadPods, nil
+	return sadPods, sadPodsCount == 0, nil
```

Comment:
Could you please elaborate the need for an extra boolean flag please? A brief look raises the question: doesn't len(sadPods) === sadPodsCount?

Reply:
In one of the early returns, we return a list of `sadPods` that may be empty, but it doesn't necessarily mean that the cluster is ok, as it can be in `conditions.WrongPodCount`. It's indeed a function that deserves some refactoring, as it doesn't just get sad pods.